sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs

```diff
@@ -20,5 +21,10 @@ public Uri AuthorityHost
             get { return _authorityHost ?? AzureAuthorityHosts.GetDefault(); }
             set { _authorityHost = Validations.ValidateAuthorityHost(value); }
         }
+
+        /// <summary>
+        /// If <c>true</c>, the tenant Id provided by a service authorization challenge will be used over the tenantId configured via the credential options.
```

Comment:
Is this description correct?

Reply:
Yes, although looking at it now, I think it should default to true.

Reply:
The name of the property and the description do not connect in my head :)

Reply:
Tenant discovery is accomplished via challenge responses and the tenant returned in the challenge could differ from the one passed to the TokenCredential at construction time.  Initially this was named something like `PreferTenantIdHint`. Does that help, or do you have a suggestion for how one or the other should be renamed/re-worded?

I think when we write a realistic sample for this, it might make more sense.

Reply:
What about something like:

> If <c>true</c>, enables the credential to fetch tokens for any tenants the user or multi-tenant application registration is a member of. Otherwise the credential will only acquire tokens from the tenant configured when the credential was constructed.